### PR TITLE
Support GLES2 for iOS

### LINKED ISSUE #22271: OSIphone doesn't support GLES2 as video driver yet
**Godot version:**
Master (e20a28a7)

**OS/device including version:**
iOS

**Issue description:**
This is likely an oversight and not the sign of any particular technical difficulty, but the current iOS platform code doesn't have the toggle for GLES2 as a video driver, so it still supports only GLES3 as in Godot 3.0:

https://github.com/godotengine/godot/blob/08f452d1a95d7f171472c33a60983d95adf1665d/platform/iphone/os_iphone.cpp#L52-L60

This should be easy to fix for any iOS contributor by copying the way other platforms initialize their video driver (see in particular 08f452d1).

### PR DESCRIPTION
**Pull request by volzhs**
Fix #22271
Tested on iPhone 4s

note that I need to delete `default_env.tres` file not to crash.

### CODE REVIEW

File: platform/iphone/os_iphone.cpp [hunk 1]
```diff
@@ -117,7 +157,8 @@ Error OSIPhone::initialize(const VideoMode &p_desired, int p_video_driver, int p
 	//visual_server->cursor_set_visible(false, 0);
 
 	// reset this to what it should be, it will have been set to 0 after visual_server->init() is called
-	RasterizerStorageGLES3::system_fbo = gl_view_base_fb;
+	if (use_gl3) RasterizerStorageGLES3::system_fbo = gl_view_base_fb;
```
Comment (akien-mga) [inline platform/iphone/os_iphone.cpp]: `clang-format` doesn't like this :)
https://travis-ci.org/godotengine/godot/jobs/466521381

**Comment by akien-mga** [comment]
CC @samgreen @hpvb

**Comment by samgreen** [comment]
I’ll do a more thorough review and testing later today but this looks good so far. Thanks for doing this!

**Comment by akien-mga** [comment]
Merging for now, we can review more in-depth later on if needed, but it looks good to me too from a quick glance.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**